src/bblocks_data_importers/world_bank/wb_api.py

```diff
@@ -0,0 +1,429 @@
+        config: Optional[dict] = None,
+    ) -> None:
+        """Fetch a World Bank indicator and transform it into a cleaned DataFrame.
+
+        Args:
+            series str | list[str]: The World Bank indicator series code(s).
+            config Optional[dict]: Configuration for the data fetch.
+
+        Returns:
+            None, the data is stored in the object.
+
+        """
+        logger.info(f"Fetching World Bank data for series: {series}")
+
+        # Set the configuration
+        if config is not None:
+            self.config.update(config)
+
+        # Store the configuration in the raw data
+        self._raw_data["config"] = self.config
+
+        # Fetch the indicator data from World Bank API, clean and structure it
+        self._raw_data["data"] = wbgapi.data.FlatFrame(
+            series=series,
+            db=self.config["database"],
+            economy=self.config["economies"],
+            time=self.config["years"],
+            mrnev=self.config["most_recent_non_empty_value"],
+            mrv=self.config["most_recent_value"],
+            params=self.config["api_params"],
+            skipBlanks=True,
+            labels=False,
+        ).reset_index()
+
+        logger.info("Data successfully fetched from World Bank API")
+
+    def clear_cache(self) -> None:
+        """Clear the data cached in the importer"""
+
+        self._raw_data = {}
+        self._data = None
+
+        logger.info("Cache cleared")
+
+    def get_data(
+        self, series: str | list[str], config: Optional[dict] = None
+    ) -> pd.DataFrame:
+        """Fetches and returns data for the specified indicator series.
+
+        Args:
+            series (str | list[str]): The indicator code(s) to retrieve.
+            Can be a single string or a list of strings.
+            config (Optional[dict]): Optional configuration settings for fetching data,
+            such as API parameters.
+
+        Returns:
+            pd.DataFrame: A DataFrame filtered to include only the requested indicator series.
+        """
+        # Ensure series is a list
+        series = [series] if isinstance(series, str) else series
+
+        # validate config
+        if config is None:
+            config = {}
+
+        # Check that all configuration keys are valid
+        if not set(config.keys()).issubset(self._valid_config_keys):
+            raise ValueError(
+                f"Invalid configuration keys: {set(config.keys()) - set(self._valid_config_keys)}"
+            )
+
+        # If the data is not loaded, load it
+        if self._data is None:
+            self._load_wb_series(series=series, config=config)
+            self._clean_data()
+
+        # If the data is loaded, check that the configuration is the same
+        elif self._raw_data["config"].copy() | config != self._raw_data[
+            "config"
+        ] or not set(series).issubset(self._data[Fields.indicator_code].unique()):
+            self.clear_cache()
```

Comment:
This is fine, though perhaps we shouldn't log a message that the cache is cleared here

Reply:
It is cleared, which is the correct behaviour. But here the user doesn't need to know the internal logic - so logging a message that the cache is cleared is confusing